### PR TITLE
Add beta docs for DBM Aurora cluster auto discovery feature

### PR DESCRIPTION
**Pull request by jmeunier28**
### What does this PR do? What is the motivation?

Adds instructions on how to use the new beta auto discovery feature in the datadog-agent to auto discovery Aurora clusters + replica instances and automatically instrument them with the postgres / mysql integration

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+---
+title: Automatically configure your aurora clusters for monitoring
+kind: guide
+---
+
+{{< beta-callout url="#" btn_hidden="true" >}}
+Aurora cluster auto discovery is in **beta**. If you have any feedback about this feature, please [contact support](https://docs.datadoghq.com/help/).
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md]: > [contact support](https://docs.datadoghq.com/help/)
inline link doesn't seem to work

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+{{< /beta-callout >}}
+
+This guide assumes that you have configured [Database Monitoring][1] for your Aurora hosts.
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:10]:
```suggestion
This guide assumes that you have configured [Database Monitoring][1] for your Amazon Aurora DB clusters.
```

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+## Agent Installation
+
+// TODO: Add the installation steps for the beta agent
+
+## Overview
+
+The Datadog Agent supports configuring automatic discovery and monitoring of your aurora clusters. This feature works in lieu of manually configuring the Agent by listing individual database host endpoints.
+
+This is helpful for environments where the number of endpoints can be dynamic, such as with clusters that use [Aurora Auto Scaling][6].
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:28]: Auto Scale will be the only reason that causes dynamic number of hosts right? Wonder if we should just say "This is helpful for Aurora clusters with auto-scaling enabled, automatically adjusting the number of read replicas in response to workload variations."

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+```yaml
+database_monitoring:
+  autodiscovery:
+    aurora:
+      enabled: true
+```
+
+By default, the listener will discover all aurora clusters in the account and region where the Agent is running that have the `datadoghq.com/scrape:true` tag applied. You can also configure the listener to discover clusters with specific tags.
+
+Specify custom tags for the aurora cluster discovery in the `datadog.yaml` file:
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:45]: should we link an aws doc on how to add/modify tags for aurora?

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "rds:DescribeDBClusters",
+        "rds:DescribeDBInstances"
+      ],
+      "Resource": "*"
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:84]:
```suggestion
      "Resource": "arn:aws:rds:<region>:<account>:cluster:<name>"
```

File: content/en/database_monitoring/guide/aurora_autodiscovery.md (hunk 4 above)
Comment (jmeunier28) [reply content/en/database_monitoring/guide/aurora_autodiscovery.md:45]: yes, i'll do that

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+}
+```
+
+You can also attach the [AmazonRDSReadOnlyAccess][3] policy.
+
+### Configure aurora
+
+By default, the listener discovers all aurora clusters in the account and region where the Agent is running that have the `datadoghq.com/scrape:true` tag applied. You can also configure the Agent to discover clusters with specific tags.
+
+The tags must be applied to the DB cluster (Role: `Regional clutser`). For more information on tagging RDS resources, see the [AWS documentation][7].
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:77]: > (Role: `Regional clutser`)

What does `Role` mean?

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+
+Specify custom tags for the aurora cluster discovery in the `datadog.yaml` file:
+
+```yaml
+database_monitoring:
+  autodiscovery:
+    aurora:
+      enabled: true
+      tags:
+        - "my-cluster-tag"
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:112]:
```suggestion
        - "my-cluster-tag-key:value"
```

**nitpick:** ‏just to make it clear it needs to follow the `<KEY>:<VALUE>` pattern

**Review by lu-zhengda — APPROVED** [review]

File: content/en/database_monitoring/guide/aurora_autodiscovery.md (hunk 6 above)
Comment (jmeunier28) [reply content/en/database_monitoring/guide/aurora_autodiscovery.md:77]: the role of the instance, it should be on the cluster and not the individual instances,  but maybe that's already clear?

Screenshot 2024-03-03 at 1 14 33 PM

Comment (lu-zhengda) [reply content/en/database_monitoring/guide/aurora_autodiscovery.md:77]: ahh i didn't know this is called `Role` in the console.

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "rds:DescribeDBClusters",
+        "rds:DescribeDBInstances"
+      ],
+      "Resource": "arn:aws:rds:<region>:<account>:cluster:<name>"
```
Comment (lu-zhengda) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:65]:
```suggestion
      "Resource": [
        "arn:aws:rds:<region>:<account>:cluster:*",
        "arn:aws:rds:<region>:<account>:db:*"
      ],
```

Comment (lu-zhengda) [reply content/en/database_monitoring/guide/aurora_autodiscovery.md:65]: the listener scrapes all clusters and dbs so `*` is required here

Comment (jmeunier28) [reply content/en/database_monitoring/guide/aurora_autodiscovery.md]: yes, thanks 🙇

File: content/en/database_monitoring/guide/_index.md [hunk 9]
```diff
@@ -17,6 +17,7 @@ cascade:
     {{< nextlink href="database_monitoring/guide/heroku-postgres" >}}Setting up Heroku Postgres for Database Monitoring{{< /nextlink >}}
     {{< nextlink href="database_monitoring/guide/tag_database_statements" >}}Tagging Database Statements{{< /nextlink >}}
     {{< nextlink href="database_monitoring/guide/managed_authentication" >}}Connecting with Managed Authentication{{< /nextlink >}}
+    {{< nextlink href="database_monitoring/guide/aurora_autodiscovery" >}}Configured Database Monitoring for your Amazon Aurora DB Clusters{{< /nextlink >}}
```
Comment (cswatt) [inline content/en/database_monitoring/guide/_index.md:20]:
```suggestion
    {{< nextlink href="database_monitoring/guide/aurora_autodiscovery" >}}Configuring Database Monitoring for your Amazon Aurora DB Clusters{{< /nextlink >}}
```

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 10]
```diff
@@ -0,0 +1,244 @@
+---
+title: Automatically configure your aurora clusters for monitoring
+kind: guide
+---
+
+{{< beta-callout url="#" btn_hidden="true" >}}
+Aurora cluster auto discovery is a beta feature. If you have any feedback about this feature, contact Support at support@datadoghq.com.
```
Comment (cswatt) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:7]:
```suggestion
Autodiscovery for Aurora clusters is a beta feature. If you have any feedback about this feature, contact support at support@datadoghq.com.
```

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+{{< /beta-callout >}}
+
+This guide assumes that you have configured Database Monitoring for your Amazon Aurora [Postgres][1] or [MySQL][11].
```
Comment (cswatt) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:10]:
```suggestion
This guide assumes that you have configured Database Monitoring for your Amazon Aurora [Postgres][1] or [MySQL][11] databases.
```

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+
+## Before you begin
+
+Supported databases
+: Postgres, MySQL
+
+Supported Agent versions
+: 7.53.0+ (Not yet released)
+
+In order to use this feature now, you need to install the beta version of the Agent. See the installation instructions below.
```
Comment (cswatt) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:20]:
```suggestion
To use this feature, you need to install the beta version of the Agent. See the [Agent installation](#agent-installation) section on this page.
```

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+## Overview
+
+The Datadog Agent supports configuring automatic discovery and monitoring of your Aurora clusters. This feature works in lieu of manually configuring the Agent by listing individual database host endpoints.
+
+This is helpful for Aurora clusters with [auto-scaling enabled][6], automatically adjusting the number of read replicas in response to workload variations.
+
+This features leverages the Agent [Autodiscovery feature][4] and lets you define configuration templates for Postgres or MySQL checks, and specify which clusters each check should apply to.
```
Comment (cswatt) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:28]:
```suggestion
Datadog's [Autodiscovery][4] enables you to configure monitoring in dynamic infrastructures. You can use this feature to monitor your Aurora clusters without having to list individual database host endpoints. This is especially helpful for clusters that use [Aurora Auto Scaling][6], which dynamically adjusts the number of Aurora Replicas in response to variations in connectivity or workload.

With Autodiscovery and Database Monitoring, you can define configuration templates for Postgres or MySQL checks and specify which clusters to apply each check to.
```

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+
+## Enabling Aurora cluster discovery
```
Comment (cswatt) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:30]:
```suggestion
## Enabling Autodiscovery for Aurora clusters
```

**Review by cswatt — COMMENTED** [review]
first batch of edits

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 15]
```diff
@@ -0,0 +1,245 @@
+---
+title: Automatically configure your aurora clusters for monitoring
```
Comment (cswatt) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:2]: I'm dithering on a good title for this page @jhgilbert

File: content/en/database_monitoring/guide/aurora_autodiscovery.md [hunk 16]
```diff
@@ -0,0 +1,245 @@
+---
+title: Configuring Database Monitoring for Amazon Aurora DB Clusters
```
Comment (jhgilbert) [inline content/en/database_monitoring/guide/aurora_autodiscovery.md:2]: @ceciliawatt I agree with your sentiments around avoiding too much focus on autodiscovery. Autodiscovery is the how, not the what. Thoughts on this title? I took it straight from the guide link.

Comment (jmeunier28) [reply content/en/database_monitoring/guide/aurora_autodiscovery.md:2]: That looks good to me

**Pull request merged**